src/main/java/io/druid/segment/loading/LoadSpec.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2015 Metamarkets Group, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.druid.segment.loading;
+
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+
+import java.io.File;
+
+/**
+ * A means of pulling segment files into a destination directory
+ */
+@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, property = "type")
+public interface LoadSpec
```

Comment:
Since the LoadSpec and the pullers are intimately tied together, I didn't define any subtypes here, instead opting to register the subtypes whenever the pullers are registered. LoadSpec is passed as a `Map<String, Object>` until it is finally ready to be consumed. At that point both the LoadSpec  subtype and the appropriate puller will be registered.

Also, I didn't want to add a bunch of dependencies into the API jar, which would be required to properly couple the actual LoadSpec impl and the SegmentPuller